lib/srv/uacc/uacc_linux.go

```diff
@@ -270,8 +275,8 @@ func LogFailedLogin(btmpPath, username, hostname string, remote [4]int32) error
 
 	accountDb.Lock()
 	defer accountDb.Unlock()
-
-	status, errno := C.uacc_add_btmp_entry(cBtmpPath, cUsername, cHostname, &cIP[0], secondsElapsed, microsFraction)
+	var uaccPathErr [uaccPathErrMaxLength]C.char
```

Comment:
Huh, interesting. I thought you'd have to use `C.malloc` here, but I guess this may work too...?

Reply:
it will because that memory is going to be part of the heap allocation that makes up the stack for the goroutine which is bound to the CGO call anyway.